src/utils/node_byte_utils.ts

```diff
@@ -159,5 +160,9 @@ export const nodeJsByteUtils = {
     return nodeJsByteUtils.toLocalBufferType(buffer).write(source, byteOffset, undefined, 'utf8');
   },
 
-  randomBytes: nodejsRandomBytes
+  randomBytes: nodejsRandomBytes,
+
+  swap32(buffer: Uint8Array): NodeJsBuffer {
+    return nodeJsByteUtils.toLocalBufferType(buffer).swap32();
```

Comment:
You should be able to just `Buffer.prototype.swap32.call(buffer);` this

Reply:
I would do that for all the functions in this file, but we use the explicit `toLocalBufferType` because of this:

```txt
> Buffer.prototype.toString.call(new Uint8Array([61]))
Uncaught TypeError: this.utf8Slice is not a function
```

It seems like swap32 to works so we don't necessarily have to blanket avoid calling prototype methods like you suggest but I'm wary that Node core may feel entitled to only make swap32 work _only_ for Buffer instances. Am I being overly cautious?

Reply:
Well ... you add another typed array instance that way, but it's probably no big deal.

I'd honestly consider just fixing `.toString()` and adding tests to Node.js core to ensure that all of the methods work on Uint8Arrays myself, maybe I'll do that if I find a piece of extra time 😂

Reply:
> Well ... you add another typed array instance that way

Yea and I'm sure our whole library could get a nice perf boost if we removed all the unnecessary view creations. 


I can make this change for swap32 if we're confident it won't break but now I'm thinking maybe I file a ticket for a perf improvement where we feature detect the ability to skip over the `toLocalBufferType` for the potential future node versions where this is fixed for `toString` and tested all the others. (more indirection!).

There is cost added to startup time but we can "test" `prototype.X.call`-style calls don't throw and then lock in versions of the functions that do that instead 🤔

Reply:
https://jira.mongodb.org/browse/NODE-6550